File: dsa-05-main/python/day03/stack.py

class Stack:

	def __init__(self, size):
		self.arr = [0] * size
		self.top = -1

	def push(self, val):
		self.top = self.top + 1
		self.arr[self.top] = val

	def pop(self):
		self.top = self.top - 1

	def peek(self):
		return self.arr[self.top]

	def isEmpty(self):
		return self.top == -1

	def isFull(self):
		return self.top == len(self.arr)-1


def main():
	s = Stack(6)
	s.push(11)
	s.push(22)
	s.push(33)
	s.push(44)

	val = s.peek()
	print("Element on top = " + str(val) + "\n")

	while not s.isEmpty():
		val = s.peek()
		s.pop()
		print("Popped : " + str(val))


if __name__ == "__main__":
	main()

File: dsa-05-main/cpp/day03/stack.cpp
#include <iostream>
using namespace std;

#define SIZE	6

class my_stack {
private:
	int arr[SIZE];
	int top;
public:
	my_stack() {
		top = -1;
		for (int i = 0; i < SIZE; i++)
			arr[i] = 0;
	}
	void push(int ele) {
		top++;
		arr[top] = ele;
	}
	void pop() {
		top--;
	}
	int peek() {
		return arr[top];
	}
	bool is_empty() {
		return top == -1;
	}
	bool is_full() {
		return top == (SIZE - 1);
	}
};

int main() {
	my_stack s;
	int option, val;
	do {
		cout << "0. exit\n1. push\n2. peek\n3. pop\nenter option: ";
		cin >> option;
		switch (option) {
		case 1: // push
			if (s.is_full())
				cout << "stack full." << endl;
			else {
				cout << "enter ele: ";
				cin >> val;
				s.push(val);
			}
			break;
		case 2: // peek
			if (s.is_empty())
				cout << "stack empty." << endl;
			else {
				val = s.peek();
				cout << "next ele : " << val << endl;
			}
			break;
		case 3: // pop
			if (s.is_empty())
				cout << "stack empty." << endl;
			else {
				val = s.peek();
				s.pop();
				cout << "popped ele : " << val << endl;
			}
			break;
		}
	} while (option != 0);
	return 0;
}

